src/librustc/infer/error_reporting/mod.rs

```diff
@@ -357,7 +359,7 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
                 // for imported and non-imported crates
                 if exp_path == found_path
                 || exp_abs_path == found_abs_path {
-                    let crate_name = self.tcx.crate_name(did1.krate);
+                    let crate_name = self.tcx.cstore().crate_name_untracked(did1.krate);
```

Comment:
Why can't you use `crate_name` here?

Reply:
This is a spurious change that should not have happened. I introduced it after hastily rebasing to HEAD of master when the moving of `cstore` from `sess` to `tcx` happened. I'll revert.